### DISCUSSION TITLE
what does peer-[:nth-of-type(1)]  means?

### DISCUSSION THREAD

**Discussion by A-caibird** [tailwindlabs/tailwindcss#11794] [Help] [answered]
Below is my html code,When my mouse hovers over c, why does the font color of p2 also turn red?
```
 <div class="bg-[yellowgreen] w-full h-full peer/edit1">. // a
            <p>
                hello-edit1
            </p>
        </div>
        <div class="bg-[yellowgreen] w-full h-full peer/edit1"> // b
            <p>
                hello-edit12
            </p>
        </div>
        <div class="bg-[yellowgreen] w-full h-full peer/edit2"> // c
            <p>
                hello-edit2
            </p>
        </div>
        <div class="bg-[yellowgreen] w-full h-full peer/edit2"> // e
            <p>
                hello-edit21
            </p>
        </div>
        <div class="peer-[:nth-of-type(1)_&:hover]/edit1:text-[red] ]"> // p1
            <p class="">第一个</p>
        </div>
        <div class="peer-[:nth-of-type(2)_&:hover]/edit2:text-[blue] ]"> //p2
            <p class="">第二个</p>
        </div>
```

**Comment by A-caibird** [comment]
turn blue, sorry,why does the font color of p2 also turn blue?

**Reply by A-caibird** [reply]
[this Tailwind Play,why it is hidden](https://play.tailwindcss.com/ORHYXqplqM)

**Reply by wongjn** [reply]
With regards to the font color turning blue, I was unable to replicate the unexpected behavior you describe in [Tailwind Play](https://play.tailwindcss.com/c5FlIM2Mn8), which suggests there could be some other misconfiguration/misuse. Consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces the unexpected behavior.

As for "why it is hidden", assuming you are referencing the `<div>` element by "it", it is because the element has [`display: none`](https://developer.mozilla.org/en-US/docs/Web/CSS/display#none) applied to it (which "turns off the display of an element"[^1]) via the `hidden` utility class.

[^1]: https://developer.mozilla.org/en-US/docs/Web/CSS/display#none

**Reply by A-caibird** [reply]
When my mouse hovers over c( <div class="bg-[yellowgreen] w-full h-full peer/edit2"> // c
            <p>
                hello-edit2
            </p>
        </div>), why does not text color of( <div class="peer-[:nth-of-type(2)_&:hover]/edit2:text-[blue] ]"> //p2
            <p class="">第二个</p>
        </div>)  turn blue?

**Reply by wongjn** [reply, marked as answer]
The `peer-[:nth-of-type(2)_&:hover]/edit2:` variant does not match since the parent of any previous sibling `peer/edit2` elements is not the second child of type (as per `:nth-of-type(2)_&`).

**Reply by A-caibird** [reply]
can you give  a example use  peer-[:nth-of-type(2)_&] ? thank you very much

**Reply by wongjn** [reply]
https://play.tailwindcss.com/IA73kFj9NZ

**Reply by A-caibird** [reply]
Thank you very much. I still have a little problem: [:nth-of-type(2)_&], I go through your example, think this variant is to choose if his outermost parent element has two siblings of the same type element, is this correct? My [example detailed](https://play.tailwindcss.com/IA73kFj9NZ)


Why is :nth-of-type(2) in css different? https://developer.mozilla.org/zh-CN/docs/Web/CSS/:nth-of-type

**Reply by wongjn** [reply]
I believe what you may be confused with is that the arbitrary variant `[:nth-of-type(2)_&]` has [descendent combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator), thus the parent assertion. Compare this with `[&:nth-of-type(2)]` which has no such combinator and thus more closely aligns with the "vanilla" behavior of `:nth-of-type(2)`: https://play.tailwindcss.com/Y3016xKbqD

**Reply by A-caibird** [reply]
can you give an example about [:nth-of-type(2)_&]  [&:nth-of-type(2)]   [:nth-of-type(2)] ?thank you

**Reply by wongjn** [reply]
`[:nth-of-type(2)_&]`: https://play.tailwindcss.com/IA73kFj9NZ
`[&:nth-of-type(2)]`: https://play.tailwindcss.com/Y3016xKbqD
`[:nth-of-type(2)]`: https://play.tailwindcss.com/PpPYcRs9Et

**Reply by A-caibird** [reply]
thank you,i understand over
